### PR TITLE
Refactor build logic

### PR DESCRIPTION
**Pull request by aasitnikov**
Pick up from #89 

In this PR:
* Rewrite build.gradle file to kts
* Rewrite buildSrc classes to Kotlin
* Bump versions of Kotlin, AGP and Gradle
* Add sources jar to aar modules (#44)
* Publish android modules using AGP built-in maven-publish
* Remove generation of BuildConfig classes in android modules

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        remove(repo)
+        addFirst(repo)
+    }
+}
+
+subprojects {
+    apply(plugin = "checkstyle")
+
+    tasks.register<Checkstyle>("checkstyle") {
+        description = "Runs Checkstyle inspection against ICanPlayer sourcesets."
```
Comment (alaershov) [inline build.gradle.kts]: What is `ICanPlayer`?

File: buildSrc/src/main/kotlin/ProjectConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    const val javapoet = "com.squareup:javapoet:1.11.1"
+    const val junit = "junit:junit:4.12"
+    const val mockito = "org.mockito:mockito-core:2.27.0"
+    const val truth = "androidx.test.ext:truth:1.1.0"
+    const val compiletesting = "com.google.testing.compile:compile-testing:0.17"
+    const val asm = "org.ow2.asm:asm:7.1"
+    const val asmUtil = "org.ow2.asm:asm-util:7.1"
+    const val autoservice = "com.google.auto.service:auto-service:1.0-rc5"
+    const val autocommon = "com.google.auto:auto-common:0.10"
+    const val kotlinStdlib = "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlinVersion"
```
Comment (alaershov) [inline buildSrc/src/main/kotlin/ProjectConfiguration.kt]: Maybe just `stdlib`?

File: buildSrc/src/main/kotlin/ProjectConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    const val kotlinStdlibForCompiler = "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion"
+    const val coroutines = "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.1"
```
Comment (alaershov) [inline buildSrc/src/main/kotlin/ProjectConfiguration.kt]: also maybe bump coroutines to 1.3.8?

File: moxy-android/stub-android/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+plugins {
+    java
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_7
```
Comment (alaershov) [inline moxy-android/stub-android/build.gradle.kts]: why 1.7 here and 1.8 in other modules? Maybe use 1.8 everywhere?

**Review by alaershov — APPROVED** [review]
Nice job! Some minor comments, address them if you want.

File: build.gradle.kts (hunk 1 above)
Comment (aasitnikov) [reply build.gradle.kts]: I guess it was copypasted from somewhere :)

File: buildSrc/src/main/kotlin/ProjectConfiguration.kt (hunk 2 above)
Comment (aasitnikov) [reply buildSrc/src/main/kotlin/ProjectConfiguration.kt]: Actually removed this dependency altogether, kotlin gradle plugin 1.4.0 adds stdlib dependency automatically

File: buildSrc/src/main/kotlin/ProjectConfiguration.kt (hunk 3 above)
Comment (aasitnikov) [reply buildSrc/src/main/kotlin/ProjectConfiguration.kt]: Yeah, good point

File: moxy-android/stub-android/build.gradle.kts (hunk 4 above)
Comment (aasitnikov) [reply moxy-android/stub-android/build.gradle.kts]: It doesn't really matter, as this modules are used just as stubs. But bumped to 1_8 for consistency.

**Pull request merged**